Sources/BulletinViewController.swift

```diff
@@ -337,6 +338,58 @@ extension BulletinViewController: UIViewControllerTransitioningDelegate {
 
 }
 
+// MARK: - Keyboard
+
+extension BulletinViewController {
+    func setUpKeyboardLogic() {
+        NotificationCenter.default.addObserver(self, selector: #selector(onKeyboardShow), name: .UIKeyboardWillShow, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(onKeyboardHide), name: .UIKeyboardWillHide, object: nil)
```

Comment:
You should add a `deinit` block in the declaration of `BulletinViewController` to unregister the observer and prevent retain cycles:

~~~swift
deinit {
    NotificationCenter.default.removeObserver(self)
}
~~~

Reply:
This isn't strictly required by Foundation framework as of iOS 9: https://useyourloaf.com/blog/unregistering-nsnotificationcenter-observers-in-ios-9/